### PR TITLE
[JBPM-9310] Warning in BAMTaskEventListener if task is skipped from t…

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9310

### CODE REVIEW

**Comment by elguardian** [comment]
depends on 
https://github.com/kiegroup/droolsjbpm-knowledge/pull/460
https://github.com/kiegroup/jbpm/pull/1741
https://github.com/kiegroup/droolsjbpm-integration/pull/2211

**Comment by elguardian** [comment]
Jenkin retest this please.

**Comment by elguardian** [comment]
Jenkin retest this please.

**Comment by elguardian** [comment]
Jenkin retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by elguardian** [comment]
Jenkins restest this please.

**Comment by elguardian** [comment]
failure is not related. anyway trying again a clean run.
Jenkins retest this please.

**Review by afalhambra — APPROVED** [review]
Looks good to me. Nice work @elguardian

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 1]
```diff
@@ -708,12 +709,19 @@ protected void addAsyncHandler(final KModuleDeploymentUnit unit, final InternalK
...
+            ObjectModel model = new ObjectModel(
                             "mvel",
                             "new org.jbpm.services.task.lifecycle.listeners.BAMTaskEventListener(false)"
-                    )
             );
+            if(!oldList.contains(model)) {
+                if(!oldList.isEmpty()) {
+                    oldList.add(0, model);
+                } else {
+                    oldList.add(model);
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:721]: even if the list is empty, add(0, model) would add it in the first position, is this if-else worth it?

**Review by gmunozfe — COMMENTED** [review]
Looks good, just some minor comments

**Review by gmunozfe — COMMENTED** [review]
Looks good, just a couple of minor comments

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:721]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 2]
```diff
@@ -101,6 +101,7 @@
 import org.kie.internal.runtime.conf.NamedObjectModel;
 import org.kie.internal.runtime.conf.ObjectModel;
 import org.kie.internal.runtime.conf.RuntimeStrategy;
+import org.kie.internal.runtime.manager.deploy.DeploymentDescriptorImpl;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java:104]: done

**Pull request merged**